### ISSUE TITLE
bug in rodrigues interface

### ISSUE THREAD

**Issue by TariqBlecher** [ska-sa/simulator#2]
Hi, When I tried to run a sim through rodrigues a couple days ago, it crashed .. some error about not finding an MS

**Comment by sphemakh** [comment]
Could you include the log, or the exact error.

**Comment by TariqBlecher** [comment]
2015/06/10 16:22:52 WARNING: MS/smakh1433953364.213687.MS doesn't exist
2015/06/10 16:22:52 INFO: redirecting log output to
/storage/29-ygw1msac/output/plots-smakh1433953364.213687/log-smakh1433953364.213687.txt
2015/06/10 16:22:52 PYXIS: creating directory
/storage/29-ygw1msac/output/plots-smakh1433953364.213687
2015/06/10 16:22:52 ERROR: Exception raised, aborting
2015/06/10 16:22:52 ERROR: Exception raised, aborting
2015/06/10 16:22:52 ERROR: exiting with error code 1, check the logs for
errors
2015/06/10 16:22:52 INFO: last logfile was
/storage/29-ygw1msac/output/plots-smakh1433953364.213687/log-smakh1433953364.213687.txt
2015/06/10 16:22:52 INFO: elapsed time is 0h0m8.14s
simulation crashed (state 1)

**Comment by sphemakh** [comment]
can I see the contents of this file: /storage/29-ygw1msac/output/plots-smakh1433953364.213687/log-smakh1433953364.213687.txt

**Comment by TariqBlecher** [comment]
2015/06/10 16:26:43 INFO: ## Wed Jun 10 16:26:35 2015:
/usr/bin/pyxis CFG=/storage/30-jqml7aub/input/parameters.json
OUTFILE=//storage/30-jqml7aub/output/results
OUTDIR=/storage/30-jqml7aub/output azishe
Traceback (most recent call last):
File "/usr/bin/pyxis", line 281, in <module>
Pyxis.Internals.run(*commands);
File "/usr/lib/python2.7/dist-packages/Pyxis/Internals.py", line 1098, in
run
comcall();
File "./pyxis-ckat.py", line 60, in azishe
ms.plot_uvcov(ms=.1,width=10,height=10,dpi=150,save="$OUTFILE-uvcov.png")
File "/usr/lib/python2.7/dist-packages/Pyxides/ms.py", line 230, in
plot_uvcov
tab = ms(msname);
File "/usr/lib/python2.7/dist-packages/Pyxides/ms.py", line 66, in ms
tab = table(msname,readonly=not write,ack=False);
File "/usr/lib/python2.7/dist-packages/pyrap/tables/table.py", line 319, in
**init**
Table.**init** (self, tabname, lockopt, opt);
RuntimeError: Table MS/smakh1433953595.423050.MS does not exist
2015/06/10 16:26:43 ERROR: Exception raised, aborting

**Comment by sphemakh** [comment]
Check that you are creating the MS properly in your pyxis-script. There should be a `log-simms.txt`, that might shed some light on what went wrong. Post this log here if you not sure what's going wrong. This partly my fault, I should find a way of raising an exception in simms if something goes wrong in CASAPY when creating the MS.

**Comment by TariqBlecher** [comment]
Well this is actually your script, not sure why it suddenly stopped working

**Comment by sphemakh** [comment]
Not sure how to respond.

**Comment by o-smirnov** [comment]
Well @TariqBlecher you do need to post the logfiles that Sphe is asked for. It may be his script but he's not psychic!

**Comment by TariqBlecher** [comment]
haha yeah I posted both log files, anyway here is the uncut version

2015/06/10 16:26:35 INFO: ## Wed Jun 10 16:26:35 2015:
/usr/bin/pyxis CFG=/storage/30-jqml7aub/input/parameters.json
OUTFILE=//storage/30-jqml7aub/output/results
OUTDIR=/storage/30-jqml7aub/output azishe
2015/06/10 16:26:35 PYXIS: running command azishe
2015/06/10 16:26:35 PYXIS: registered module 'im.lwimager'
2015/06/10 16:26:35 PYXIS: executing '/bin/mkdir MS':

# 

The start-up time of CASA may vary
depending on whether the shared libraries

# are cached or not.

casaviewer: could not start Xvfb
94-casaviewer.wrapped-svr: cannot connect to X server
CASA Version 4.2.2 (r30986)
Compiled on: Thu 2014/09/04 17:59:42 UTC

---

For help use the following commands:
tasklist - Task list organized by category
taskhelp - One line summary of available tasks
help taskname - Full help for task
toolhelp - One line summary of available tools
help par.parametername - Full help for parameter name

---

will execute script /tmp/tmpbNZknS.py

---

I am going to upgrade your configuration in:
/root/.casa/ipython
Initializing from configuration:
/opt/casapy-42.2.30986-1-64b/lib/python2.7/site-packages/IPython/UserConfig

Successful upgrade!

All files in your directory:
/root/.casa/ipython
which would have been overwritten by the upgrade were backed up with a .old
extension. If you had made particular customizations in those files you may
want to merge them back into the new files.
Please press <RETURN> to start IPython.

---

Activating auto-logging. Current session state plus future input saved.
Filename : ipython-20150610-162643.log
Mode : backup
Output logging : False
Raw input log : False
Timestamping : False
State : active
2015-06-10 16:26:43 INFO casa::::casa CASA Version 4.2.2 (prerelease r30986)
2015-06-10 16:26:43 INFO casa::::casa Tagged on: Thu, 21 Aug 2014
**\* Loading ATNF ASAP Package...
**\* ... ASAP (4.2.0a rev#30794) import complete ***
##################################################

Major interface changes to SINGLE DISH tasks have been
taken place in CASA 4.2.2 release

The interface of the following tasks are modified:
sdbaseline, sdcal, sdcal2, sdfit, sdflag, sdgrid,
sdimaging, sdmath, sdplot, sdreduce, sdsave, and sdstat.
Additionally, a new task called sdaverage is available. Task
sdsmooth has been incorporated in the new task and removed.

The tasks with old interfaces are available with name
{taskname}old. They will be kept until CASA 4.3 release
and removed from later releases. Users are advised to
update existing scripts.

##################################################

> > > > 4 0.25
> > > > [0;31m---------------------------------------------------------------------------[0m
> > > > [0;31mSystemExit[0m Traceback (most recent call last)

[0;32m/opt/casapy-42.2.30986-1-64b/lib/python2.7/casapy.py[0m in
[0;36m<module>[0;34m()[0m
[0;32m----> 1[0;31m [0;34m[0m[0m
[0m[1;32m 2[0m [0;34m[0m[0m
[1;32m 3[0m [0;34m[0m[0m
[1;32m 4[0m [0;34m[0m[0m
[1;32m 5[0m [0;34m[0m[0m

[0;32m/opt/casapy-42.2.30986-1-64b/lib/python2.7/casapy.py[0m in
[0;36m<module>[0;34m()[0m
[1;32m 1[0m [0;31m# Auto Gen casapy script. From simms.py[0m
[0;34m[0m[0m
[1;32m 2[0m
[0;30mexecfile[0m[0;34m([0m[0;34m"/code/simms/casasm.py"[0m[0;34m)[0m[0;34m[0m[0m
[0;32m----> 3[0;31m
[0;30mmakems[0m[0;34m([0m[0;30mmsname[0m[0;34m=[0m[0;34m"MS/
smakh1433953595.423050.MS"[0m[0;34m,[0m
[0;30mlabel[0m[0;34m=[0m[0;34m"None"[0m[0;34m,[0m
[0;30mtel[0m[0;34m=[0m[0;34m"meerkat"[0m[0;34m,[0m
[0;30mpos[0m[0;34m=[0m[0;34m"observatories/MeerKAT64_ANTENNAS"[0m[0;34m,[0m
[0;30mpos_type[0m[0;34m=[0m[0;34m"casa"[0m[0;34m,[0m
[0;30mra[0m[0;34m=[0m[0;34m"0h0m0s"[0m[0;34m,[0m
[0;30mdec[0m[0;34m=[0m[0;34m"-30d0m0s"[0m[0;34m,[0m
[0;30msynthesis[0m[0;34m=[0m[0;36m0.25[0m[0;34m,[0m
[0;30mscan_length[0m[0;34m=[0m[0;36m4[0m[0;34m,[0m
[0;30mdtime[0m[0;34m=[0m[0;34m"10s"[0m[0;34m,[0m
[0;30mfreq0[0m[0;34m=[0m[0;34m"700MHz"[0m[0;34m,[0m
[0;30mdfreq[0m[0;34m=[0m[0;34m"50MHz"[0m[0;34m,[0m
[0;30mnchan[0m[0;34m=[0m[0;36m1[0m[0;34m,[0m
[0;30mstokes[0m[0;34m=[0m[0;34m"LL LR RL RR"[0m[0;34m,[0m
[0;30mstart_time[0m[0;34m=[0m[0;34m-[0m[0;36m2[0m[0;34m,[0m
[0;30msetlimits[0m[0;34m=[0m[0;30mFalse[0m[0;34m,[0m
[0;30melevation_limit[0m[0;34m=[0m[0;36m0.000000[0m[0;34m,[0m
[0;30mshadow_limit[0m[0;34m=[0m[0;36m0.000000[0m[0;34m,[0m
[0;30mcoords[0m[0;34m=[0m[0;34m"itrf"[0m[0;34m,[0m[0;30mlon_lat[0m[0;34m=[0m[0;30mNone[0m[0;34m,[0m
[0;30mnoup[0m[0;34m=[0m[0;30mFalse[0m[0;34m)[0m[0;34m[0m[0m
[0m[1;32m 4[0m [0;30mexit[0m[0;34m[0m[0m
[1;32m 5[0m [0;34m[0m[0m

[0;32m/opt/casapy-42.2.30986-1-64b/lib/python2.7/casapy.py[0m in
[0;36mmakems[0;34m(msname, label, tel, pos, pos_type, ra, dec, synthesis,
scan_length, dtime, freq0, dfreq, nchan, start_time, stokes, noise,
setlimits, elevation_limit, shadow_limit, mspath, coords, lon_lat, noup)[0m
[1;32m 55[0m [0;30mscan_lenght[0m [0;34m=[0m
[0;30msynthesis[0m[0;34m[0m[0m
[1;32m 56[0m [0;32mprint[0m
[0;34m'>>>>'[0m[0;34m,[0m[0;30mscan_length[0m[0;34m,[0m[0;30msynthesis[0m[0;34m[0m[0m
[0;32m---> 57[0;31m
[0;30msys[0m[0;34m.[0m[0;30mexit[0m[0;34m([0m[0;34m)[0m[0;34m[0m[0m
[0m[1;32m 58[0m [0;30mstart_time[0m [0;34m=[0m [0;30mstart_time[0m
[0;32mor[0m
[0;34m-[0m[0;30mscan_length[0m[0;34m/[0m[0;36m2[0m[0;34m[0m[0m
[1;32m 59[0m [0;34m[0m[0m

[0;31mSystemExit[0m:
Type %exit or %quit to exit IPython (%Exit or %Quit do so unconditionally).
leaving casapy...
2015/06/10 16:26:35 PYXIS: /bin/mkdir succeeded
2015/06/10 16:26:35 INFO: lwimager version is 1.3.2
DONE: simms.py succeeded. MS is at MS/smakh1433953595.423050.MS
2015/06/10 16:26:43 WARNING: MS/smakh1433953595.423050.MS doesn't exist
2015/06/10 16:26:43 INFO: redirecting log output to
/storage/30-jqml7aub/output/plots-smakh1433953595.423050/log-smakh1433953595.423050.txt
2015/06/10 16:26:43 PYXIS: creating directory
/storage/30-jqml7aub/output/plots-smakh1433953595.423050

**Comment by sphemakh** [comment]
I'm shutting this shit down. Come back when your senses return to you.

**Comment by o-smirnov** [comment]
Hang on, does this make sense?

```
DONE: simms.py succeeded. MS is at MS/smakh1433953595.423050.MS
2015/06/10 16:26:43 WARNING: MS/smakh1433953595.423050.MS doesn't exist
```

**Comment by o-smirnov** [comment]
I.e. does look like a bug to me, but @TariqBlecher, please try to be more specific in your bug reports, the beginning of this thread was very vague and unhelpful.

**Comment by TariqBlecher** [comment]
Apologies, lack of sleep must be getting to me

**Comment by sphemakh** [comment]
You are probably passing unicode into `simms()` convert all "strings" (which are actually unicode) to proper strings.  Below is how I deal with this and other potential issues that may arise from the json parameter file.

```
    with open(II(config)) as jsn_std:
        params = json.load(jsn_std)

    for key in params.keys():
        if params[key] =="":
            del params[key] 
        elif isinstance(params[key],unicode):
            params[key] = str(params[key])
```

see https://github.com/ska-sa/simulator/blob/master/src/pyxis-ckat.py lines 81-86

**Comment by TariqBlecher** [comment]
sorry Sphe there's been a miscommunication. I'm not referring to a bug in
my pipeline. I just ran a default observation on rodrigues with the
simulator form and it crashed so I just wanted to let you know in case you
were not aware.

**Comment by sphemakh** [comment]
Wow, I see. I'll have a look at it.

**Comment by sphemakh** [comment]
You re-deployed RODDRIGUES on rodrigues.meqtrees.net, and in the process pre-maturely committed some experimental changes in the simulator container. Once you done with meqsilhouette demos, we will need to discuss how new containers and their forms should be added to rodrigues.meqtrees.net.

**Comment by gijzelaerr** [comment]
@SpheMakh I don't get what happened here. was RODRIGUES deployed or the simulation container? Also, how did experimental changes end up in the master? You should assume the latest master is the deployed version, or will be at some point. I'm planning on some auto update mechanism that will update RODRIGUES as soon as you push to the github master.

**Comment by sphemakh** [comment]
RODRIGUES was deployed. And it fetched the simulation container from github, the container had fixes from that imaging in stellenbosch conference which were done in haste, and I hadn't had a chance to test/review them properly.

**Comment by gijzelaerr** [comment]
RDORIGUES is being deployed all the time, we have been fixing issues with
it.

I updated all the containers on the machine also to the latest on the
docker hub. Story short, If you don't want to have changes in your
container you should not push them to the master. Those containers are
automatically build and can be deployed to the machine when you update the
containers.

2015-06-11 11:47 GMT+02:00 Sphesihle Makhathini notifications@github.com:

> RODRIGUES was deployed. And it fetched the simulation container from
> github, the container had fixes from that imaging in stellenbosch
> conference which were done in haste, and I hadn't had a chance to
> test/review them properly.
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/ska-sa/simulator/issues/2#issuecomment-111069920.

## 

Gijs Molenaar
http://pythonic.nl

**Comment by o-smirnov** [comment]
Yeah so I think it's just a question of being disciplined about what goes
in the master, correct?

**Comment by sphemakh** [comment]
Yep @o-smirnov

**Comment by sphemakh** [comment]
fixed in 6410948
